lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,41 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      final titleString = nodes[0].text;
+      RegExp regex = RegExp(r'\b\d+\b');
+      Iterable<RegExpMatch> matches = regex.allMatches(titleString);
+      List<String> numbers = matches.map((match) => match.group(0)).toList();
```

Comment:
```suggestion
      RegExp xxxRule = RegExp(r'\b\d+\b');
      Iterable<RegExpMatch> xxx = xxxRule.allMatches(titleString);
      List<String> firstOfxxx = matches.map((match) => match.group(0)).toList();
```

Suggest providing meaningful name to these places. The above change suggestion are just an example.
Complicated operation like these regexp finding and matching may usually cause confusing for future developer.
For example: what does `r'\b\d+\b'` mean and why ??? (Of course I also suggest you explain the reason why you choose this regexp in the pr description)
In fact, the current TAT project contains lots of these bad practice, which use non-meaningful variable or function name. So in order not to add more this kind of code, we may consider making these improvements as much as possible.

Reply:
[d4388a5](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/d4388a5388e36d5335a0f36db666a49ccda3c94b)
[8f5b5b4](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/8f5b5b43fc4bdfad3e6ca32ec86d0e68270c3b57)
The title String of the first course table was stored seperately in its <td> element,
but it currently stores all the information in a row, 
(ex: 學號：110310144　　姓名：xxx　　班級：電機三甲　　　 112 學年度 第 1 學期　上課時間表)
so the RegExp is used to filter out only the number parts
and the "numbers" should consist of three numerical values (ex: [110310144, 112, 1])

Reply:
@James-Lu-none Could you please add this into PR description ?

Reply:
ok

Reply:
Since you have made changes in [ab8c839](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/ab8c839e6cae3f30e62e1fdb0845f23daafad480) and [8c9b6ab](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/8c9b6abfa25c62da2b2949d0519ae26e9445b6b6), suggest rewrite the description you just added.
In addition, suggest adding what exact regexp it is, since you mentioned "the Regexp".

Reply:
ok

Reply:
In fact, I hope that these contents can be written in "Description" rather than in "Additional note". 

The biggest difference between a PR and an issue is that an issue describes what happened, while a PR describes how you solved the problem, not just re-describing it.

On this detail, the Regexp you used is crucial because it critically filters out the information we urgently need. Therefore, suggest clearly explaining these in the Description, as it is precisely the method you used to solve the problem.

Reply:
ok